File: python/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        row = [1]
        for i in range(rowIndex):
            new_row = [0]
            for j in range(i + 1):
                new_row[j] += row[j]
                new_row.append(row[j])
            row = new_row
        return row

File: rust/pascals-triangle-ii.rs
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut row: Vec<i32> = vec![1];
        for i in 1..(row_index + 1) as usize {
            let mut new_row: Vec<i32> = vec![0];
            for j in 0..i {
                new_row[j] += row[j];
                new_row.push(row[j]);
            }
            row = new_row;
        }
        return row;
    }
}
